__tests__/users.ts

```diff
@@ -2047,6 +2143,264 @@ describe('mutation updateUserProfile', () => {
+      'https://twitter.com/lee/',
+    ];
+    const invalid = [
+      'lee#',
+      'http://twitter.com/lee',
+      'http://x.com/lee',
+      'http://twitter.com',
+      'http://x.com',
+      'twitter.com',
+      'x.com',
+      'https://example.com/u/lee',
+      'https://twitter.com/lee?bla=1',
+      'https://x.com/lee?bla=1',
+    ];
+
+    valid.forEach((item) => {
+      expect(twitterSocialUrlMatch.test(item)).toBe(true);
+      expect(item.match(twitterSocialUrlMatch)?.groups?.value).toBe('lee');
+    });
+
+    invalid.forEach((item) => {
+      expect(twitterSocialUrlMatch.test(item)).toBe(false);
+    });
+  });
+
+  it('should validate roadmap handle', () => {
+    const valid = [
+      'lee',
+      'roadmap.sh/u/lee',
+      'https://roadmap.sh/u/lee',
+      'https://roadmap.sh/u/lee/',
+    ];
+    const invalid = [
+      'lee#',
+      'http://roadmap.sh/lee',
+      'http://roadmap.sh',
+      'roadmap.sh',
+      'https://example.com/u/lee',
+      'https://roadmap.sh/lee?bla=1',
+    ];
+
+    valid.forEach((item) => {
+      expect(roadmapShSocialUrlMatch.test(item)).toBe(true);
+      expect(item.match(roadmapShSocialUrlMatch)?.groups?.value).toBe('lee');
+    });
+
+    invalid.forEach((item) => {
+      expect(roadmapShSocialUrlMatch.test(item)).toBe(false);
+    });
+  });
+
+  it('should validate threads handle', () => {
+    const valid = [
+      'lee',
+      'threads.net/lee',
+      'https://threads.net/@lee',
+      'https://threads.net/lee',
+      'https://threads.net/lee/',
+    ];
+    const invalid = [
+      'lee#',
+      'http://threads.net/lee',
+      'http://threads.net',
+      'threads.sh',
+      'https://example.com/u/lee',
+      'https://threads.net/lee?bla=1',
+    ];
+
+    valid.forEach((item) => {
+      expect(threadsSocialUrlMatch.test(item)).toBe(true);
+      expect(item.match(threadsSocialUrlMatch)?.groups?.value).toBe('lee');
+    });
+
+    invalid.forEach((item) => {
+      expect(threadsSocialUrlMatch.test(item)).toBe(false);
+    });
+  });
+
+  it('should validate codepen handle', () => {
+    const valid = [
+      'lee',
+      'codepen.io/lee',
+      'https://codepen.io/lee',
+      'https://codepen.io/lee/',
+    ];
+    const invalid = [
+      'lee#',
+      'http://codepen.io/lee',
+      'http://codepen.io',
+      'codepen.sh',
+      'https://example.com/u/lee',
+      'https://codepen.io/lee?bla=1',
+    ];
+
+    valid.forEach((item) => {
+      expect(codepenSocialUrlMatch.test(item)).toBe(true);
+      expect(item.match(codepenSocialUrlMatch)?.groups?.value).toBe('lee');
+    });
+
+    invalid.forEach((item) => {
+      expect(codepenSocialUrlMatch.test(item)).toBe(false);
+    });
+  });
+
+  it('should validate reddit handle', () => {
+    const valid = [
+      'lee',
+      'reddit.com/u/lee',
+      'reddit.com/user/lee',
+      'https://reddit.com/u/lee',
+      'https://reddit.com/user/lee',
+      'https://reddit.com/user/lee/',
```

Comment:
Would `https://reddit.com/u/lee/` be valid as well?

Reply:
not sure if we intend to test all valid use cases, if yes examples like this are missing in some tests

Reply:
I could have missed ones I will check, I did not really test all the valid ones (too many) but the most common ones users could paste into the box. 

> Would https://reddit.com/u/lee/ be valid as well?

yes, I'll add!

Reply:
Added in 617ed0d749301ffd7bbc6d573d885da618e34455